packages/nestjs/src/setup.ts

```diff
@@ -67,35 +67,47 @@ export { SentryTracingInterceptor };
  */
 class SentryGlobalFilter extends BaseExceptionFilter {
   public readonly __SENTRY_INTERNAL__: boolean;
+  private readonly _logger: Logger;
 
   public constructor(applicationRef?: HttpServer) {
     super(applicationRef);
     this.__SENTRY_INTERNAL__ = true;
+    this._logger = new Logger('ExceptionsHandler');
   }
 
   /**
    * Catches exceptions and reports them to Sentry unless they are expected errors.
    */
   public catch(exception: unknown, host: ArgumentsHost): void {
-    if (isExpectedError(exception)) {
-      return super.catch(exception, host);
+    // The BaseExceptionFilter does not work well in GraphQL applications.
+    // By default, Nest GraphQL applications use the ExternalExceptionFilter, which just rethrows the error:
+    // https://github.com/nestjs/nest/blob/master/packages/core/exceptions/external-exception-filter.ts
+    if (host.getType<'graphql'>() === 'graphql') {
+      // neither report nor log HttpExceptions
+      if (exception instanceof HttpException) {
+        throw exception;
+      }
+
+      if (exception instanceof Error) {
+        this._logger.error(exception.message, exception.stack);
+      }
+
+      captureException(exception);
+      throw exception;
+    }
+
+    if (!isExpectedError(exception)) {
+      captureException(exception);
     }
 
-    captureException(exception);
     return super.catch(exception, host);
   }
 }
 Catch()(SentryGlobalFilter);
 export { SentryGlobalFilter };
 
 /**
- * Global filter to handle exceptions and report them to Sentry.
- *
- * The BaseExceptionFilter does not work well in GraphQL applications.
- * By default, Nest GraphQL applications use the ExternalExceptionFilter, which just rethrows the error:
- * https://github.com/nestjs/nest/blob/master/packages/core/exceptions/external-exception-filter.ts
- *
- * The ExternalExceptinFilter is not exported, so we reimplement this filter here.
+ * Global filter to handle exceptions in NestJS + GraphQL applications and report them to Sentry.
  */
 class SentryGlobalGraphQLFilter {
```

Comment:
should we deprecate this one then?

Reply:
First we wanna release it, then at some point deprecate.